pkg/router/template/template_helper.go

```diff
@@ -376,17 +376,29 @@ func parseIPList(list string) string {
 		return ""
 	}
 
+	var validIPs []string
+
 	ipList := strings.Fields(list)
 	for _, ip := range ipList {
-		if net.ParseIP(ip) == nil {
-			if _, _, err := net.ParseCIDR(ip); err != nil {
-				log.V(7).Info("parseIPList found not IP/CIDR item", "value", ip, "err", err)
-				return ""
-			}
+		// check if it's a valid IP
+		if net.ParseIP(ip) != nil {
+			validIPs = append(validIPs, ip)
+		} else if _, _, err := net.ParseCIDR(ip); err == nil {
+			// check if it's a valid CIDR
+			validIPs = append(validIPs, ip)
+		} else {
+			// Log invalid IP/CIDR
+			log.V(0).Info("parseIPList found invalid IP/CIDR", ip)
```

Comment:
Since this changes behavior, we need to make sure this error is noticed, and indicate that the invalid IP is now ignored.  Something like this:
```suggestion
			log.Error("error parsing IP list, ignoring invalid IP/CIDR", ip)
```

Reply:
I added a customErr for IP/CIDR since only `net.parseCIDR` returns `err`.

Reply:
I'm wondering whether this one should be an error, taking into account that now we don't stop the parsing and there can be some valid IPs still. I think that the error would make more sense down the code where we find that the list is empty. However in the similar conditions in the same function (when we return `""`) we use `log.V(7).Info()`. So it seems to be inconsistent.

Reply:
@alebedev87  That’s why I use log.V(0).Info at the beginning.
I reviewed all scenarios that call log.Error() in `template_helper.go`, and they all return "" immediately.
Since, in this case, we will continue parsing the rest of the IP, is it okay to use log.Info but at a higher log level?
If yes, what level should I use here?

Reply:
[The default log level is 2](https://github.com/openshift/router/blob/master/images/router/haproxy/Dockerfile.ocp#L22) so if we want this message to be visible it has to be `<= 2`. From what I can see in `template_helper.go` the log level `0` is what's used in similar cases ([example](https://github.com/openshift/router/blob/40db90072b633e6d0ec7d064bd8441f12065a2c4/pkg/router/template/template_helper.go#L136)) - when the user needs to be informed but the flow may continue. Similarly, looking at the other functions - failing the parsing with an error if the list of valid IPs/CIDRs is empty seems to be the way to go.

Reply:
Btw the condition about the trimmer list being compared to the given list doesn't seem to use the right logging level:
```go
	if trimmedList != list {
		log.V(7).Info("parseIPList leading/trailing spaces found")
		return ""
	}
```

I would prefer it to be a level `0` too as we want the user to see this.